stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -311,16 +325,16 @@ public class MessageListViewModel(
 
                             newState.copy(
                                 newMessageState = newMessageState,
-                                unreadCount = getUnreadMessageCount(newMessageState)
+                                unreadCount = getUnreadMessageCount(newMessageState),
                             )
                         } else {
                             newState
                         }
 
-                        messagesState.messageItems.firstOrNull {
-                            it is MessageItemState && it.message.id == scrollToMessage?.id
-                        }?.let {
-                            focusMessage((it as MessageItemState).message.id)
+                        if (messagesState.scrollToPositionState == LoadFocusedMessageData && scrollToMessage != null) {
+                            messagesState.messageItems.firstOrNull {
+                                it is MessageItemState && it.message.id == scrollToMessage?.id
+                            }?.let { focusMessage((it as MessageItemState).message.id) }
```

Comment:
do we have to have this check? can't we attempt to focusMessage(scrollToMessage.id) and if it's null, nothing happens?

Reply:
We do not, but to me it does not make sense to iterate through the whole list each time data updates if we do not have to.